Jenkinsfile

```diff
@@ -470,355 +453,395 @@ def testGo(prefix, packagesToTest) {
+      ],
+      'github.com/keybase/client/go/kbfs/libpages': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/kbfs/libpages/config': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/kbfs/search': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/kbfs/simplefs': [
+        flags: '-race',
+        timeout: '2m',
+      ],
+      'github.com/keybase/client/go/kbfs/test': [
+        name: 'kbfs_test_race',
+        flags: '-race',
+        timeout: '12m',
+      ],
+      'github.com/keybase/client/go/kbfs/tlf': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+      'github.com/keybase/client/go/kbfs/tlfhandle': [
+        flags: '-race',
+        timeout: '30s',
+      ],
+    ],
+    test_windows_go_: [
+      '*': [],
+      'github.com/keybase/client/go/systests': [
+        disable: true,
+      ],
+      'github.com/keybase/client/go/chat': [
+        disable: true,
+      ],
+    ],
+  ]
+  def getOverallTimeout = { testSpec ->
+    def timeoutMatches = (testSpec.timeout =~ /(\d+)([ms])/)
+    return [
+      time: 1 + (timeoutMatches[0][1] as Integer),
+      unit: timeoutMatches[0][2] == 's' ? 'SECONDS' : 'MINUTES',
+    ]
+  }
+  def defaultPackageTestSpec = { pkg ->
+    def dirPath = pkg.replaceAll('github.com/keybase/client/go/', '')
+    def testName = dirPath.replaceAll('/', '_')
+    return [
+      name: testName,
+      flags: '',
+      timeout: '30m',
+      dirPath: dirPath,
+    ]
+  }
+  def getPackageTestSpec = { pkg ->
+    if (testSpecMap[prefix].containsKey(pkg)) {
+      if (testSpecMap[prefix][pkg]) {
+        def testSpec = testSpecMap[prefix][pkg]
+        if (testSpec['disable']) {
+          return false
         }
-        return defaultPackageTestSpec(pkg)
-      }
-      if (testSpecMap[prefix].containsKey('*')) {
-        return defaultPackageTestSpec(pkg)
+        return defaultPackageTestSpec(pkg) + testSpec
       }
-      return false
+      return defaultPackageTestSpec(pkg)
+    }
+    if (testSpecMap[prefix].containsKey('*')) {
+      return defaultPackageTestSpec(pkg)
+    }
+    return false
+  }
+  stage("Setup Test Suite") {
+  packagesToTest.each { pkg, _ ->
+    def testSpec = getPackageTestSpec(pkg)
+    if (!testSpec) {
+      return
     }
-    packagesToTest.each { pkg, _ ->
-      def testSpec = getPackageTestSpec(pkg)
-      if (!testSpec) {
-        return
-      }
-
-      println "Running go vet for ${pkg}"
-      sh "go vet ${pkg} || (ERR=\$? && echo \"go vet failed with error code \$ERR\" && exit \$ERR)"
-
-      if (isUnix()) {
-        // Windows `gofmt` pukes on CRLF, so only run on *nix.
-        println "Check that files are formatted correctly"
-        sh "test -z \$(gofmt -l \$(sed 's/github.com.keybase.client.go.//' ${pkg} ))"
-      }
 
+    def testBinary = "${testSpec.name}.test"
+    def test = {
```

Comment:
Hmm so you're parallelizing the compilation now too? I was worried about this since it'll result in a bunch of parallel `go test -c` processes theoretically clobbering the same dependency packages.

Reply:
what do you mean by clobbering? all of the deps are already vendored right?

Reply:
I mean during compilation, a bunch of go processes will be racing to build a bunch of dependency code on top of each other, doing extra work.

Reply:
got it. you're probably right but i think it's fine (or at least it's faster).